app/controllers/application_controller.rb

```diff
@@ -45,21 +45,166 @@ def setup_breadcrumb_navigation
+        add_practice_breadcrumb(practice)
+      # If there are any duplicate breadcrumbs, delete them
+      elsif practice_breadcrumb(practice).present? && practice_breadcrumb(practice).count > 1
+        remove_breadcrumb(practice_breadcrumb(practice))
+        add_practice_breadcrumb(practice)
+      end
+    end
+
+    # practice checklist path
+    if params[:action] == 'planning_checklist' && params[:controller] == 'practices'
+      practice = get_practice_by_practice_id
+
+      empty_breadcrumbs
+      add_practice_breadcrumb(practice)
+      session[:breadcrumbs] << { 'display': 'Planning checklist', 'path': practice_planning_checklist_path(practice) }
+    end
+
+    # practice committed path
+    if params[:action] == 'committed' && params[:controller] == 'practices'
+      practice = get_practice_by_practice_id
+
+      empty_breadcrumbs
+      add_practice_breadcrumb(practice)
+      add_checklist_breadcrumb(practice)
+      session[:breadcrumbs] << { 'display': 'Confirmation', 'path': practice_committed_path(practice) }
+    end
+
     # practice partners path
     if params[:action] == 'index' && params[:controller] == 'practice_partners'
       # empty the bread crumbs and start a new 'path'
-      session[:breadcrumbs] = []
-      session[:breadcrumbs] << {'display': 'Partners', 'path': practice_partners_path}
+      empty_breadcrumbs
+      session[:breadcrumbs] << { 'display': 'Partners', 'path': practice_partners_path }
     end
 
     # practice partner show path
     if params[:action] == 'show' && params[:controller] == 'practice_partners'
       # add the practice partner to the crumbs if it's not there already
       @practice_partner = PracticePartner.friendly.find(params[:id])
-      unless session[:breadcrumbs].find {|b| b['display'] == @practice_partner.name}.present?
-        session[:breadcrumbs] << {'display': @practice_partner.name, 'path': practice_partner_path(@practice_partner)}
+      practice = Practice.find_by(slug: session[:user_return_to].split('/').pop) if session[:user_return_to].present?
+      partner_breadcrumb = session[:breadcrumbs].find { |b| b['display'] == @practice_partner.name }
+      add_partner_breadcrumb = session[:breadcrumbs] << { 'display': @practice_partner.name, 'path': practice_partner_path(@practice_partner) }
+
+
+      if practice.present? && practice_breadcrumb(practice).blank?
+        add_practice_breadcrumb(practice)
+      end
+
+      if partner_breadcrumb.blank?
+        add_partner_breadcrumb
+      # If there are any duplicate practice partner name breadcrumbs, delete them
+      elsif partner_breadcrumb.present? && partner_breadcrumb.count > 1
+        remove_breadcrumb(partner_breadcrumb)
+        add_partner_breadcrumb
       end
     end
+
+    ### PRACTICE EDITOR BREADCRUMBS ###
+    # Instructions breadcrumbs
+    if params[:action] == 'instructions' && params[:controller] == 'practices'
+      practice = get_practice_by_practice_id
```

Comment:
I'm  wondering if we can refactor lines 149-152 out since we repeat several times below

Reply:
Let me know what you think!

Reply:
that is what I was thinking! thank you josh!